jdaviz/configs/cubeviz/plugins/spectral_extraction/spectral_extraction.py

```diff
@@ -579,6 +579,18 @@ def extract(self, return_bg=False, add_data=True, **kwargs):
         pix_scale_factor = self.cube.meta.get('PIXAR_SR', 1.0)
         spec.meta['_pixel_scale_factor'] = pix_scale_factor
 
+        # Follows user-selected flux unit for extracted spectrum, ignoring bg_spec.
+        uc = self.app.get_tray_item_from_name("g-unit-conversion")
+        if uc.flux_unit_selected and (uc.flux_unit_selected != spec.flux.unit.to_string()):
+            # While specutils has with_flux_unit method, it does not handle uncertainty unit,
+            # so we roll our own here, copying internals from that method.
+            eqv = u.spectral_density(spec.spectral_axis)
+            new_spec_flux = spec.flux.to(uc.flux_unit_selected, equivalencies=eqv)
+            spec._data = new_spec_flux.value
+            spec._unit = new_spec_flux.unit
+            if spec.uncertainty:
+                spec.uncertainty.unit = new_spec_flux.unit
```

Comment:
is it possible to do this without converting the units themselves?  This risks propagating errors (although it would be limited to 2 conversions, so might not be a major risk if there isn't an alternative).

Reply:
@cshanahan1 suggested "i think that there needs to be unit conversion done in the plugin to convert the final result from the data units to the selected unit." Did I misunderstand?

ref: https://jira.stsci.edu/browse/JDAT-4734?focusedId=715315&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-715315